src/common/guac_surface.c

```diff
@@ -435,8 +447,11 @@ static int __guac_common_surface_should_use_webp(guac_common_surface* surface,
     if (!guac_client_supports_webp(surface->client))
         return 0;
 
-    /* Usage criteria are currently the same as JPEG */
-    return __guac_common_surface_should_use_jpeg(surface, rect);
+    /* Calculate the average framerate for the given rect */
+    int framerate = __guac_common_surface_calculate_framerate(surface, rect);
+
+    /* WebP is preferred if framerate is high enough */
+    return framerate >= GUAC_COMMON_SURFACE_JPEG_FRAMERATE;
```

Comment:
Please remove this change. This is not within the scope of [GUAC-1365](https://glyptodon.org/jira/browse/GUAC-1365).

> WebP should always be superior, therefore we're not checking the PNG optimality as we do for JPEG.

We do not yet have sufficient evidence to make such a claim. I know you have seen nothing but amazing performance out of WebP on your end, but our own results do not match, and so long as we are not using lossless WebP, there will always be a need for PNG.

If this change were made as-is, scrolling through a terminal window, a high-contrast document, etc. will rapidly turn ugly and (possibly) painfully slow.